client/lib/signup/cart.js

```diff
@@ -11,7 +11,33 @@ var wpcom = require( 'lib/wp' ),
 	cartValues = require( 'lib/cart-values' ),
 	cartItems = cartValues.cartItems;
 
+function addProductsToCart( cart, newCartItems ) {
+	forEach( newCartItems, function( cartItem ) {
+		cartItem.extra = Object.assign( cartItem.extra || {}, {
+			context: 'signup'
+		} );
+		const addFunction = cartItems.add( cartItem );
+
+		cart = cartValues.fillInAllCartItemAttributes( addFunction( cart ), productsList.get() );
+	} );
+
+	return cart;
+}
+
 module.exports = {
+	createCart: function( cartKey, newCartItems, callback ) {
```

Comment:
I didn't try to replace this functionality because I'm not really familiar with the signup framework and there might be things I overlooked. I have noticed in `signup/step-actions.js` functions like `addPlanToCart` which suggest that we can add item incrementally to the cart.